components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -786,6 +810,26 @@ class GeckoEngineSession(
         }
     }
 
+    @VisibleForTesting
+    // Some site add extra information after the mimetype, for example 'application/pdf; qs=0.001'
+    // we just want to extract the mimeType and ignore the rest
+    internal fun extractMimeType(contentType: String?): String? {
+        return if (contentType != null) {
+            if (contentType.contains(";")) {
+                contentType.substringBefore(";")
+            } else {
+                contentType
+            }
+        } else {
+            null
+        }
+    }
+
+    @VisibleForTesting
+    internal fun WebMessage.getHeaderValue(key: String): String? {
+        return headers[key]?.trim() ?: headers[key.toLowerCase(Locale.ROOT)]?.trim()
+    }
```

Comment:
I'm confused by this, why do we need to try this again with lowercase?

Reply:
I found cases where headers where with capital letters like `Content-Disposition` and others with lowercase `content-disposition`. I think we can just put both in lowercase but it will required that we generate another header object with all the keys in lowercase.

Reply:
@snorp Do you have any advice how Gecko handles this?

Reply:
Gecko seems to avoid putting the headers into a map, so it doesn't have this exact problem. They use `nsIHttpChannel::getResponseHeader()`, which performs case-insensitive lookup. We should probably do something similar in `WebMessage`.

Reply:
@pocmo  on the new updated gv, `headers`propperty has changed its implementation details to be a`TreeMap` [1] to avoid this issue. Should we remove our code as they are already doing it for us ? But the [public API is still a Map ](https://mozilla.github.io/geckoview/javadoc/mozilla-central/index.html) the change was only on the [implementation details level](https://hg.mozilla.org/mozilla-central/diff/0b410b21eb37f125158648d392713546fc869109/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/WebMessage.java#l1.27).

[1] `Replaced android.util.ArrayMap with java.util.TreeMap in WebMessage to enable case-insensitive handling of the HTTP headers. (bug 1666013)`

Reply:
Yeah, I think I am in favor of removing this code on our side.

Reply:
👍

Reply:
After all the behavior is in the public docs of GV now. :)